### PR TITLE
Finishmerchandisebeginmanagement

### PR DESCRIPTION
**Pull request by timrobuck**

### CODE REVIEW

File: Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            PageFactory.InitElements(BaseTest.Driver, this);
+        }
+
+        //MerchTool Loginpage elements
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.header > h1")]
+        public IWebElement PageTitle { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#UserName")]
+        public IWebElement UserName { get; set; }
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs]: InputUserName

File: Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        [FindsBy(How = How.CssSelector, Using = "#Password")]
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs]: InputPassword

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeCategoriesPage.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        {
+            ActionSelectNumberOfLines();
+            Console.WriteLine("Verify the number of entries dropdown works.");
+            ActionSortAscendingDescendingByTableColumnHeaderClick();
+            Console.WriteLine("Verify Sort mechanism on each column header works.");
+            //todo: UpdateExistigProductTypesExclusions.Click  do this only after stage is available
+        }
+
+        private void VerifyUniqueElementsExist()
+        {
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeCategoriesPage.cs]: loop through a list

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        public IWebElement HiddenIdRequired { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.row > div > fieldset:nth-child(4) > legend")]
+        public IWebElement LegendEdit { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.row > div > fieldset:nth-child(4) > i")]
+        public IWebElement InstructionNoteForSaveChangesButton { get; set; }
+
+        //column headers
+        [FindsBy(How = How.CssSelector, Using = "#FitAndFillConfigurationsForm > table > thead > tr > td:nth-child(1)")]
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs]: Are these column header locators named "ColumnHeader#" intentionally to be generic? If page specific, what's keeping us from using their names?  ColumnHeaderCrop; ColumnHeaderBleed; ColumnHeaderBuffer; etc?

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        public IWebElement InstructionNoteForTheDropDown { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.row > div > fieldset:nth-child(3) > legend")]
+        public IWebElement LegendSelect { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#productTypeId")]
+        public IWebElement DropDownProductType { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.row > div > fieldset:nth-child(3) > form > div > input")]
+        public IWebElement ButtonLoadProductType { get; set; }
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs]: Btn for naming

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditPtnCategoryPage.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+        [FindsBy(How = How.CssSelector, Using = "#SmartProductCategories > tbody > tr:nth-child(1) > td:nth-child(4) > label")]
+        private IWebElement CheckBoxRemoveRecord { get; set; } //do not test this until stage is ready.
+
+        //table record content
+        [FindsBy(How = How.CssSelector, Using = "#SmartProductCategories_info")]
+        public IWebElement ShowingEntries { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#SmartProductCategories > thead > tr > td:nth-child(1) > div")]
+        public IWebElement Column1Header { get; set; }
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditPtnCategoryPage.cs]: intentionally generic names?

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpePtnExclusionPage.cs [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            ActionSelectNumberOfLines();
+            Console.WriteLine("Verify the number of entries dropdown works.");
+            ActionSortAscendingDescendingByTableColumnHeaderClick();
+            Console.WriteLine("Verify Sort mechanism on each column header works.");
+            //todo: UpdateExistigProductTypesExclusions.Click  do this only after stage is available
+        }
+
+        private void VerifyUniqueElementsExist()
+        {
+            LableShowEntries.Verify();
```
Comment (Actiondan2857) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpePtnExclusionPage.cs]: loop?

File: Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseColorsPage.cs [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0 > tbody > tr:nth-child(2)")]
+        private IWebElement ColorTableSelectWhite { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0 > tbody > tr:nth-child(1) > td.sorting_1")]
+        private IWebElement GetTableIdValueFromFirstRow { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0 > thead > tr > td:nth-child(1) > div > span")]
+        private IWebElement SortAscendingDescendingByTableColumnHeaderIdClickTheTriangle { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.row > div.span9.view-container > fieldset > div > a")]
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseColorsPage.cs]: Not sure what you are referring to. Could you clarify for me. Thanks.

File: Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseSalesChannel_DetailsPage.cs [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        public void VerifyElements()
+        {
+            IdLabel.Verify();
+            IdText.Verify();
+            IdContent.Verify();
+            NameLabel.Verify();
+            NameContent.Verify();
+            NameSubText.Verify();
+            _commonPage.VerifyElements();
+            Console.WriteLine("Verify Elements on " + GetType().Name);
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseSalesChannel_DetailsPage.cs]: Done! Changed all the verify to Verified in the console writelines.

File: Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseSizePage.cs [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0_info")]
+        private IWebElement ShowingEntries { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0_paginate")]
+        private IWebElement BtnSetPagination { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0 > tbody > tr:nth-child(2)")]
+        private IWebElement SizeTableSelectSmall { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#DataTables_Table_0 > thead > tr > td:nth-child(1) > div > span")]
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MerchandisePages/MtMerchandiseSizePage.cs]: Done.

File: Autodan/src/pages/MerchTool/MtCommonSitePages/MtHomePageObject.cs [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.header > h1")]
+        private IWebElement MerchMgtPageTitle { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "#logoutForm > a")]
+        private IWebElement BtnLogout { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.navbar > div.navbar-inner > ul > li:nth-child(1) > a")]
+        private IWebElement NavBtnHome { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.navbar > div.navbar-inner > ul > li:nth-child(2) > a")]
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MtCommonSitePages/MtHomePageObject.cs]: Done

File: Autodan/src/pages/MerchTool/MtCommonSitePages/MtHomePageObject.cs [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+        private IWebElement NavBtnMerchandise { get; set; }
+
+        [FindsBy(How = How.CssSelector, Using = "body > div.container > div.navbar > div.navbar-inner > ul > li:nth-child(3) > a")]
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MtCommonSitePages/MtHomePageObject.cs]: Done

File: Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs (hunk 2 above)
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MtCommonSitePages/MtLoginPageObject.cs]: Don't know what you want here. Please give me a bit more. Thanks

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs (hunk 4 above)
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditFitNCropPage.cs]: It is intentional.  Generic for simplicity as column# is obvious and specific names does not seem to add anything in this case. However, I am open, tell me what you are concerned about.

File: Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditPtnCategoryPage.cs (hunk 6 above)
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/SmartProductEnginePages/MtSpeEditPtnCategoryPage.cs]: Yes, intentional as same as above. I did change to ColumnHeader1 instead to be consistent.

File: Autodan/src/pages/MerchTool/MerchandisePages/MtCommonToMerchandisePages.cs [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+        public void RunCommonActions()
+        {
+            ActionExportToCsvButton(); 
+            ActionSelectNumberOfLines(); 
+            ActionSortAscendingDescendingByTableColumnHeaderClick();
+            
+        }
+        private void ActionExportToCsvButton()
+        {
```
Comment (timrobuck) [inline Autodan/src/pages/MerchTool/MerchandisePages/MtCommonToMerchandisePages.cs]: I removed the action of clicking on this button. Only verifying that it is there. If you think I should do the click and download, lets talk about this. I can put it into my next, and last smaller branch.

**Comment by timrobuck** [comment]
Dan, Finished the list of comments and responded to each(let me know if you don’t see my response comments).I have pushed the changes out to the branch for you to review again if you want, but still have a couple we need to talk over. These are a few that I need clarification best done in person before I can fix.
When you get back to work, lets have a short discussion, especially for the clarifications. Don’t worry about it until then, get some rest.

Hope you get to feeling better.

**Comment by timrobuck** [comment]
Dan, checked in to finish review.

File: Autodan/tests/MerchTool/MtSmokeMerchandiseTests/MtSmokeMerchandiseTest.cs [hunk 14]
```diff
@@ -95,21 +95,25 @@ public void SizePageTest()
...
         }
 
-        [Test]
-        public void ProductTypeDetailPageTest()
-        {
-            //todo: create this page class and verify elements
-        }
+        //todo:will restore this shortly
+        //[Test]
+        //public void ProductTypeDetailPageTest()
```
Comment (Actiondan2857) [inline Autodan/tests/MerchTool/MtSmokeMerchandiseTests/MtSmokeMerchandiseTest.cs]: should this be restored?

Comment (timrobuck) [inline Autodan/tests/MerchTool/MtSmokeMerchandiseTests/MtSmokeMerchandiseTest.cs]: no, this is what I commented out for the class I staged just so I would not have to burden you with extra stuffl

**Comment by Actiondan2857** [comment]
completed pr

**Pull request merged**